strawberry/fastapi/router.py

```diff
@@ -62,7 +83,11 @@ def __init__(
         self.keep_alive_interval = keep_alive_interval
         self.debug = debug
         self.root_value_getter = root_value_getter or self.__get_root_value
-        self.context_getter = context_getter or self.__get_context
+        self.context_getter = (
+            self.__get_context_getter(context_getter)
+            if context_getter is not None
+            else self.__get_context_getter(lambda: {})
+        )
```

Comment:
```suggestion
        self.context_getter = self.__get_context_getter(context_getter or lambda: {})
```

Reply:
I tried this before and was getting a syntax error. However, looking at it again with fresh eyes, it works if the `lambda` is enclosed in parentheses, so:
```python
self.context_getter = self.__get_context_getter(context_getter or (lambda: {}))
```

Reply:
Now that I think about it, would
```python
self.context_getter = self.__get_context_getter(context_getter or dict)
```
work?

Reply:
It needs to be a lambda closure as `context_getter` is a `Callable` that injects dependencies. However, I've also decided it's easier to have `context_getter or (lambda: None)` here and then `**(custom_getter or {})` on line 46. These are all requirements for the MyPy checks to pass.